scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,390 @@
+        ftpClient.disconnect();
+
+        getOut().println("Disconnected");
+
+        return 1;
+    }
+
+    private void FtpGet(FTPClient ftpClient) throws IOException, IllegalArgumentException {
+        getOut().println("Importing file(s) from " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+        FTPFile[] ftpFile = ftpClient.listFiles(ftpRemoteFilePath);
+        if (ftpFile.length == 0)
+            throw new IllegalArgumentException(ftpRemoteFilePath + " not found. Please, enter a valid path.");
+
+        // If it is a single file:
+        if (ftpFile.length == 1 && ftpRemoteFilePath.contains(ftpFile[0].getName())) {
+            String saveFilePath = Paths.get(ftpLocalFilePath, ftpFile[0].getName()).toString();
+            boolean download = downloadSingleFile(ftpClient, ftpRemoteFilePath, saveFilePath);
+            if (download) {
+                getOut().println("Succeed to download the file " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+            } else {
+                getOut().println("COULD NOT download the file: " + ftpRemoteFilePath);
+            }
+
+            // If the file is a zip, and ftpExtractArchive is set to true
+            if (ftpExtractArchive && ftpRemoteFilePath.endsWith(".zip")) {
+                extractArchive(saveFilePath);
+            }
+        } // If it is a folder, download all its contents recursively
+        else {
+            ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+            downloadDirectory(ftpClient, "", "", ftpLocalFilePath);
+        }
+        getOut().println("END Import file(s) from FTP.");
+    }
+
+    private void extractArchive(String saveFilePath) throws IOException {
+        try (ZipFile zipFile = new ZipFile(saveFilePath)) {
+            // Create an enumeration of the entries in the zip file
+            Enumeration zipFileEntries = zipFile.entries();
+            // Process each entry
+            while (zipFileEntries.hasMoreElements()) {
+                // grab a zip file entry
+                ZipEntry entry = (ZipEntry) zipFileEntries.nextElement();
+                Path path = Paths.get(ftpLocalFilePath, entry.getName());
+                getOut().println("path: " + path.toString());
+                if (entry.isDirectory()) {
+                    Files.createDirectories(path);
+                } else {
+                    Path parentDir = path.getParent();
+                    getOut().println("parentDir of path: " + parentDir.toString());
+                    if (!Files.exists(parentDir)) {
+                        Files.createDirectories(parentDir);
+                    }
+                    if (Files.exists(path)) {
+                        new File(path.toString()).delete();
+                    }
+
+                    Files.copy(zipFile.getInputStream(entry), path);
+                }
+            }
+        }
+    }
+
+    private void FtpPut(FTPClient ftpClient) throws IOException, IllegalArgumentException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)